### PR TITLE
build: only check patch diffs in testing builds

### PR DESCRIPTION
**Pull request by codebytere**
We should only be testing whether or not the patch diffs have changed in testing builds - diffs during the release process should not fail the release process itself.

cc @jkleinsc @MarshallOfSound 

Notes: none

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -256,23 +258,25 @@ step-gclient-sync: &step-gclient-sync
...
-          # Export it
-          mkdir -p ../../patches
-          git format-patch -1 --stdout --keep-subject --no-stat --full-index > ../../patches/update-patches.patch
-          echo
-          echo "======================================================================"
-          echo "There were changes to the patches when applying."
-          echo "Check the CI artifacts for a patch you can apply to fix it."
-          echo "======================================================================"
-          exit 1
+        if [ "$IS_RELEASE" == "false" ]; then
```
Comment (MarshallOfSound) [inline .circleci/config.yml:261]:
```suggestion
        if [ "$IS_RELEASE" != "true" ]; then
```

We should fail into the "not release" state

File: .circleci/config.yml [hunk 2]
```diff
@@ -101,12 +101,14 @@ machine-mac-large-arm: &machine-mac-large-arm
 env-testing-build: &env-testing-build
   GN_CONFIG: //electron/build/args/testing.gn
   CHECK_DIST_MANIFEST: '1'
+  IS_RELEASE: false
```
Comment (MarshallOfSound) [inline .circleci/config.yml:104]: Don't need this if we change the if condition below

**Review by MarshallOfSound — APPROVED** [review]

**Review by jkleinsc — APPROVED** [review]
LGTM

**Comment by codebytere** [comment]
Merging `fast-track` as the failures are not germane to this config change.

**Pull request merged**